src/core/server/http/http_config.ts

```diff
@@ -62,6 +63,14 @@ export const config = {
         ),
         schema.boolean({ defaultValue: false })
       ),
+      csp: schema.object({
```

Comment:
As @pgayvallet mentioned above, this moves the config key from `csp.*` to `server.csp.*`, which makes sense but is a breaking change. Is the plan to wait for config deprecations to land in #52251 first? If so, can you put that in a list of TODOs in the PR description so we can be sure the work needed to merge this is clear?